Core/SemanticVersioning/SemanticVersionParser.cs

```diff
@@ -34,7 +35,7 @@ public SemanticVersion ParseVersion (string version)
   {
     if (!Regex.IsMatch(version, _versionPattern, RegexOptions.Multiline))
     {
-      throw new ArgumentException("Version has an invalid format. Expected equivalent to '1.2.3' or '1.2.3-alpha.4'.");
+      throw new UserInteractionException("Version has an invalid format. Expected equivalent to '1.2.3' or '1.2.3-alpha.4'.");
```

Comment:
These parser-exceptions would be good candiates for a better solution. Maybe use a ParseException or rewrite to TryParse. Either way, further optimization here is too big for the current PR so we keep the UIEx for now. It just feels icky.